### PR TITLE
Remove LogCollector Splunk public certificate

### PR DESCRIPTION
**Pull request by hjiawei**
## Description

This feature was never officially released to our customer and was mostly tested internally before. When the public certificate isn't available in the cluster, it continuously outputs "Assuming http protocol" in operator logs. This causes confusions to our customers.

## For PR author

- [ ] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

**Comment by hjiawei** [comment]
cc @penkeysuresh .

**Review by penkeysuresh — APPROVED** [review]
LGTM with respect to splunk changes.

File: pkg/controller/logcollector/logcollector_controller.go [hunk 1]
```diff
@@ -689,41 +684,17 @@ func getSplunkCredential(client client.Client) (*render.SplunkCredential, error)
...
-
-	if err := client.Get(context.Background(), certificateNamespacedName, certificateSecret); err != nil {
-		if errors.IsNotFound(err) {
-			log.Info(fmt.Sprintf("Splunk certificate secret %v not provided. Assuming http protocol or trusted CA certificate.",
-				render.SplunkFluentdCertificateSecretName))
-		} else {
-			return nil, fmt.Errorf("Failed to read secret %q: %s", render.SplunkFluentdCertificateSecretName, err)
-		}
-	} else {
-		if certificate, ok = certificateSecret.Data[render.SplunkFluentdSecretCertificateKey]; !ok || len(certificate) == 0 {
```
Comment (rene-dekker) [inline pkg/controller/logcollector/logcollector_controller.go:718]: Can't we just add it to the trusted bundle here?

Comment (hjiawei) [reply pkg/controller/logcollector/logcollector_controller.go:718]: Yeah. We could. As this isn't really a customer request before, I would like to remove it until we get a real requirement.

**Comment by hjiawei** [comment]
> @hjiawei is this PR still in progress?

Not in progress. I have confirmed with @penkeysuresh that this feature shouldn't be used by our customer and we don't have public documentation for it either. Our KB also suggests no ssl verify for self-signed certificates. I would like to get this change merged if possible.

**Review by rene-dekker — APPROVED** [review]
lgtm

**Pull request merged**